lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectData.java

```diff
@@ -728,7 +729,14 @@ public static Schema makeNullable(Schema schema) {
     do {
       if (excludeJava && c.getPackage() != null && c.getPackage().getName().startsWith("java."))
         break; // skip java built-in classes
-      for (Field field : c.getDeclaredFields())
+      Field[] declaredFields = c.getDeclaredFields();
+      Arrays.sort(declaredFields, new Comparator<Field>() {
```

Comment:
We can also use the shorthand:
```suggestion
      Arrays.sort(declaredFields, Comparator.comparing(Field::getName));
```